src/app/core/config-setup/config-import-parser.service.spec.ts

```diff
@@ -0,0 +1,242 @@
+          id: "ignored",
+          label: "ignored",
+          dataType: null,
+        },
+        {
+          id: "name",
+          label: "name",
+          dataType: "string",
+        },
+      ],
+      [
+        {
+          name: "name",
+          schema: {
+            dataType: "string",
+            label: "name",
+          },
+        },
+      ]
+    );
+  });
+
+  it("should generate id for field from label if not given", () => {
+    expectToBeParsedIntoEntityConfig(
+      [
+        {
+          id: null,
+          label: "date of birth",
+          dataType: "string",
+        },
+      ],
+      [
+        {
+          name: "dateOfBirth",
+          schema: {
+            dataType: "string",
+            label: "date of birth",
+          },
+        },
+      ]
+    );
+  });
+
+  it("should generate sensible ids from labels", () => {
+    const labelIdPairs = [
+      ["name", "name"],
+      ["Name", "name"],
+      ["FirstName", "firstName"],
+      ["name of", "nameOf"],
+      ["test's name", "testsName"],
+      ["name 123", "name123"],
+      ["123 name", "123Name"], // this is possible in JavaScript
+    ];
+
+    for (const testCase of labelIdPairs) {
+      const generatedId = ConfigImportParserService.generateIdFromLabel(
+        testCase[0]
+      );
+      expect(generatedId).toBe(testCase[1]);
+    }
+  });
+
+  it("should generate enum from additional column and reuse if already exists", () => {
+    const parsedConfig = expectToBeParsedIntoEntityConfig(
+      [
+        {
+          id: "hometown",
+          label: "hometown",
+          dataType: "enum",
+          additional_type_details: "Berlin,Hongkong,New York,",
+        },
+        {
+          id: "city",
+          label: "city",
+          dataType: "enum",
+          additional_type_details: "Berlin,Hongkong,New York,",
+        },
+        {
+          id: "missingEnum",
+          label: "missing",
+          dataType: "enum",
+          additional_type_details: null,
+        },
+      ],
+      [
+        {
+          name: "hometown",
+          schema: {
+            dataType: "configurable-enum",
+            label: "hometown",
+            innerDataType: "hometown",
+          },
+        },
+        {
+          name: "city",
+          schema: {
+            dataType: "configurable-enum",
+            label: "city",
+            innerDataType: "hometown", // reuse the previous enum!
+          },
+        },
+        {
+          name: "missingEnum",
+          schema: {
+            dataType: "configurable-enum",
+            label: "missing",
+            innerDataType: "???", // reuse the previous enum!
+          },
+        },
+      ]
+    );
+
+    expect(parsedConfig["enum:hometown"]).toEqual([
+      { id: "Berlin", label: "Berlin" },
+      { id: "Hongkong", label: "Hongkong" },
+      { id: "New York", label: "New York" },
+      // TODO: should we rename the enum ids for distinction from labels? (--> NEW_YORK)
```

Comment:
Maybe also use the `generateIdFromLabel` function.

Reply:
The advantage of leaving the IDs as given could be that during data imports, the id probably automatically matches the imported value without any matching logic from our code.